src/Datum.ts

```diff
@@ -275,18 +280,81 @@ export function getOrd<A>(O: Ord<A>): Ord<Datum<A>> {
  * `Apply` semigroup
  *
  * @since 2.0.0
+ * 
+ * Note: Semantics changed in 4.0.0 to match new Apply instance
  */
-export const getApplySemigroup = <A>(S: Semigroup<A>): Semigroup<Datum<A>> =>
-  getSemigroup(S);
+export const getApplySemigroup = <A>(S: semigroup.Semigroup<A>): semigroup.Semigroup<Datum<A>> => ({
+  // TODO: replace with apply.getApplySemigroup if we bump the min supported version of fp-ts >= 2.10
+  // Or just remove in place of that factory instance
+  concat: (first: Datum<A>, second: Datum<A>) =>
```

Comment:
This behavior is fully generalized for any Apply and Semigroup instances and exported from the Apply module in fp-ts 2.10. I just ported that behavior here and monomorphised it slightly.